File: PROCEDURAL/ProceduralUnityTests/Assets/SpaceGene/Planet/PlanetSettings.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

[CreateAssetMenu(menuName = "Scriptable Objects/PlanetSettings")]
public class PlanetSettings : ScriptableObject {

    [Range(0.0f, 10.0f)]
    public float minRadius = 0.1f;
    [Range(0.0f, 1.0f)]
    public float radiusMultiplier = 0.2f;
    [Range(0.0f, 2.0f)]
    public float speedMultiplier = 1.0f;

    public NoiseSettings baseLayer;
    public NoiseSettings[] noiseLayers;
}

File: PROCEDURAL/CPPLib/ProcLib/PlanetSettings.cpp
#include "PlanetSettings.h"

//PLANET
float PlanetSettings::MIN_RADIUS = 0.1;
float PlanetSettings::RADIUS_MULTIPLIER = 0.2;
float PlanetSettings::SPEED_MULTIPLIER = 1.0;
//SURFACE
SurfaceNoiseSettings PlanetSettings::BASE_LAYER = SurfaceNoiseSettings();
std::vector<SurfaceNoiseSettings> PlanetSettings::ADDITIONAL_LAYERS = std::vector<SurfaceNoiseSettings>(0);
